spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BinderFactoryConfiguration.java

```diff
@@ -161,12 +166,32 @@ public CompositeMessageChannelConfigurer compositeMessageChannelConfigurer(
 	@Bean
 	public static MessageHandlerMethodFactory messageHandlerMethodFactory(CompositeMessageConverterFactory compositeMessageConverterFactory,
 								@Qualifier(IntegrationContextUtils.ARGUMENT_RESOLVERS_BEAN_NAME) HandlerMethodArgumentResolversHolder ahmar,
-								@Nullable Validator validator) {
+								@Nullable Validator validator,  ConfigurableListableBeanFactory clbf) {
+	
 		DefaultMessageHandlerMethodFactory messageHandlerMethodFactory = new DefaultMessageHandlerMethodFactory();
 		messageHandlerMethodFactory.setMessageConverter(compositeMessageConverterFactory.getMessageConverterForAllRegistered());
-		messageHandlerMethodFactory.setCustomArgumentResolvers(ahmar.getResolvers());
+		
+		/*
+		 * We essentially do the same thing as the DefaultMessageHandlerMethodFactory.initArgumentResolvers(..).
+		 * We can't do it as custom resolvers for two reasons. 
+		 * 1. We would have two duplicate (compatible) resolvers and would hope fopr the right order
+		 * 2. DefaultMessageHandlerMethodFactory.initArgumentResolvers(..) puts MessageMethodArgumentResolver 
+		 * before custom converters thus not allowing an override which kind of proves #1.
```

Comment:
I don't think "hope fopr[sic]" is the right phrase; the resolvers are in a list so the order is guaranteed.

It **is** a bit odd, though, that `PayloadAR` can be superseded, but not the others.